### DISCUSSION TITLE
Why does component-register parse the attribute's value to be non-string

### DISCUSSION THREAD

**Discussion by young122849** [solidjs/solid#738] [Q&A]
First, i was using solid-element to build a web-component. When i pass a data-user-id: "123" attribute to my custom-element, it tured out to be that the variable 'dataUserId' 's value was 123, a number not a string.

I have checked the source code of component-register and i found some code like this
![image](https://user-images.githubusercontent.com/13938315/143394476-4365591d-115d-4b2b-a720-d42de4ce38e9.png)

**Comment by ryansolid** [comment]
Hmmm...this was to parse more complicated types objects or arrays. I wrote this code like 7 years ago before it was common for libraries to support webcomponents and enabling via attribute seemed nice for declarative data in html without framework. Probably less expected now a days. I do have some libraries that depend on this behavior but could be worth considering phasing out.

**Comment by zhaoyao91** [comment]
https://github.com/ryansolid/component-register/blob/master/src/utils.ts#L114

Same problem here. This behavior is confusing and excrescent and not expected by consumers.

1. There is bug for number
2. This is more expensive since not all need this parsing
3. This is harder to understand the underlining logic than just be told: 'you just get a string!'
4. There is information losing, since, '[ [] ]' is treated as the same with '[[]]'

So I suggest to remove this parsing and leave it as it is for the users.
